tensorflow/python/ops/image_ops_impl.py

```diff
@@ -110,6 +110,26 @@ def _ImageDimensions(image):
             for s, d in zip(static_shape, dynamic_shape)]
 
 
+def _BatchImageDimensions(image):
+  """Returns the dimensions of an batch of images tensor.
+
+  Args:
+    image: A 4-D Tensor of shape `[batch, height, width, channels]`.
+
+  Returns:
+    A list of `[batch, height, width, channels]` corresponding to the dimensions of the
+    input image.  Dimensions that are statically known are python integers,
+    otherwise they are integer scalar tensors.
+  """
+  if image.get_shape().is_fully_defined():
+    return image.get_shape().as_list()
+  else:
+    static_shape = image.get_shape().with_rank(4).as_list()
+    dynamic_shape = array_ops.unpack(array_ops.shape(image), 4)
+    return [s if s is not None else d
+            for s, d in zip(static_shape, dynamic_shape)]
+
+
 def _Check3DImage(image, require_static=True):
```

Comment:
does this name make sense? This function doesn't "get an image shape" rather it:
>  Returns An empty list, if `image` has fully defined dimensions. Otherwise, a list containing an assert op is returned.